source/crc32.c

```diff
@@ -0,0 +1,72 @@
+#include <aws/cal/crc32.h>
+#include <aws/cal/hash.h>
+#include <aws/checksums/crc.h>
+
+/* hash is object, vtable is methods, hash properties (not vtable) are data?*/
+
+static void s_destroy(struct aws_hash *hash);
+static int s_update(struct aws_hash *hash, const struct aws_byte_cursor *to_hash);
+static int s_finalize(struct aws_hash *hash, struct aws_byte_buf *output);
+/* why do we need to define and implement the functions separatly? */
```

Comment:
This is a forward declaration. Without it, the vtable code below doesn't know what these functions are. Another way to write this is to define those functions here, then declare the vtable below them in the file. Either way is fine, we use both interchangeably